cEP-0018.md

```diff
@@ -0,0 +1,166 @@
+# Integration of ANTLR into coala core
+
+| Metadata |                                               |
+| -------- | --------------------------------------------- |
+| cEP      | 18                                            |
+| Version  | 1.0                                           |
+| Title    | Integration of ANTLR into coala core          |
+| Authors  | Viresh Gupta <mailto:viresh16118@iiitd.ac.in> |
+| Status   | Proposed                                      |
+| Type     | Feature                                       |
+
+## Abstract
+
+This document describes how an API based on ANTLR will be constructed and
+maintained.
+
+## Introduction
+
+ANTLR provides parsers for various language grammars and thus we can provide
+an interface from within coala and make it available to bear writers so that
+they can write advanced linting bears. This will aim at supporting the more
+flexible visitor based method of AST traversal (as opposed to listener based
+mechanisms).
+
+The proposal is to introduce a parallel concept to the coala-bears library,
+which will be called the coala-ast hereon.
+
+## Proposed Change
+
+Here is the detailed implementation stepwise:
+
+````
+1. There will be a separate repository named as coala-antlr-ast which will
+   be installable via ```pip install coala-antlr-ast```.
+2. A new package would be introduced in the coala-antlr-ast repository,
+   which would be the maintain all the dependencies for antlr-ast based
+   bears. This would be the ```antlrastlib``` package
+3. The ```antlrastlib``` package will provide endpoints relevant to the
+   visitor model of traversing the ast generated via ANTLR.
+4. Another package named ```antlrparsers``` will be created inside
+   coala-antlr-ast repository. This package will hold parsers generated
+   for various languages generated (with python target) beforehand for the
+   supported set of grammars.
+5. ```antlrastlib``` will have an AST loader class that will be responsible
+   for loading the AST of a given file depending on the extension.
+   For e.g, a .c file will be loaded using the parser from
+   ```antlrastlib``` for c language.
```

Comment:
why triple quotes inside markdown inline text?

Reply:
Thanks a lot @jayvdb 
I applied whatever patches the bear suggested and that was pretty silly...
Your suggestions worked a lot better than the bear ;)